### ISSUE TITLE
equella_list_courses_for_user&modifiable=1 returns no courses

### ISSUE THREAD

**Issue by noisyjerm** [openequella/moodle-mod_openEQUELLA#80]
**Description**
The web service returns no courses when the parameter modifiable is set to true

**To Reproduce**
Enable webservices on the Moodle instance
Add a protocol
Create a token
Add a user as a "Teacher" to a course
Visit /webservice/rest/server.php?wstoken=<TOKEN>&moodlewsrestformat=json&wsfunction=equella_list_courses_for_user&user=<USERNAME>&archived=0&modifiable=1

**Expected behavior**
The result should be an array containing courses that the user is a teacher of

**Actual result**
No data is returned

**Platform:**
- Moodle Version: 3.10
- mod_equella release 1.1 version 2020032500

**Additional context**
Perform the same steps but replace modifiable=1 with modifiable=0
Enrolled courses are returned

**Solution**
externallib.php 

Replace lines 262 to 273 with
if ($modifiable) {
    $ctx = context_course::instance($course->id);
    if (!has_capability(self::WRITE_PERMISSION, $ctx, $userobj)) {
        continue;
    }
}

Cause
line 240
courses = get_user_capability_course does not return property contextid

as required in line 264
$contextrecord->id = $course->contextid; 

using  context_course::instance($course->id) instead of eq_context_course::get_from_record($contextrecord) should be efficient as this also uses context::cache_get

**Comment by SammyIsConfused** [comment]
Hi @noisyjerm,
Thank you for taking the time to log this bug.

I have: 
- Replicated your issue - in 3.10 with the latest moodle mod, Teacher roles cannot Add To External System as the course list does not populate. 
- Replicated that setting modifiable to 0 makes the list of enrolled courses pull down properly.
- Edited my copy of the moodle-mod-openEQUELLA with your suggested changes and tested again - verified that with those changes Teacher roles CAN pull down the course list and Add To External System - but only with the courses that are enabled for that user. So that all checks out. 

So - I'll be making a pull request containing your suggested changes, with the intention that it will be included in the next released version of moodle-mod_openEQUELLA - 1.2.

Thank you so much for taking the time to debug and find this fix :)

**Comment by edalex-ian** [comment]
No, not stale. A fix has been merged and soon we hope to release an updated version of the module.
